packages/daimo-api/src/contract/coinIndexer.ts

```diff
@@ -82,42 +85,42 @@ export class CoinIndexer {
     sinceBlockNum?: bigint;
     txHashes?: Hex[];
   }): TransferOpEvent[] {
-    let ret = this.allTransfers.filter(
-      (log) => log.from === addr || log.to === addr
+    let relevantTransfers = this.allTransfers.filter(
+      (log) => log.args.from === addr || log.args.to === addr
     );
     if (sinceBlockNum) {
-      ret = ret.filter((log) => (log.blockNumber || 0n) >= sinceBlockNum);
+      relevantTransfers = relevantTransfers.filter(
+        (log) => (log.blockNumber || 0n) >= sinceBlockNum
+      );
     }
     if (txHashes !== undefined) {
-      ret = ret.filter((log) => txHashes.includes(log.txHash || "0x"));
+      relevantTransfers = relevantTransfers.filter((log) =>
+        txHashes.includes(log.transactionHash || "0x")
+      );
     }
-    // HACK: Ignore paymaster transfers for now
-    // TODO: Stop doing that and show them in UI
-    ret = ret.filter(
-      (log) =>
-        log.from !== tokenMetadata.paymasterAddress &&
-        log.to !== tokenMetadata.paymasterAddress
+
+    const transferOpsIncludingPaymaster = relevantTransfers.map((log) =>
+      this.attachTransferOpProperties(log)
     );
 
-    return ret;
+    const transferOps = this.attachFeeAmounts(transferOpsIncludingPaymaster);
+
+    return transferOps;
   }
 
-  private logToTransferOp(log: TransferLog): TransferOpEvent {
+  /* Populates atomic properties of logs to convert it Op Event.
+   * Does not account for fees since they involve multiple logs.
+   */
+  private attachTransferOpProperties(log: TransferLog): TransferOpEvent {
     const { blockNumber, blockHash, logIndex, transactionHash } = log;
     const { from, to, value } = log.args;
-    const nonceMetadata = this.opIndexer.fetchNonceMetadata(
-      transactionHash,
-      logIndex
-    );
-
-    if (
-      blockNumber == null ||
-      blockHash == null ||
-      logIndex == null ||
-      transactionHash == null
-    ) {
-      throw new Error(`pending log ${JSON.stringify(log)}`);
-    }
+    const userOp = this.opIndexer.fetchUserOpLog(transactionHash, logIndex);
```

Comment:
in-memory join (tx log > op log)

right way to do it for now, but a bit fragile--relies on `opIndexer` always updating first before we get here

that should be fine with the way `pipeLogs` currently works

we can revisit with e2pg in the future > joining happens in db or at the time of block ingestion